tm2/pkg/sdk/baseapp.go

```diff
@@ -660,14 +661,18 @@ func (app *BaseApp) runMsgs(ctx Context, msgs []Msg, mode RunTxMode) (result Res
 				fmt.Sprintf("msg:%d,success:%v,log:%s,events:%v",
 					i, false, msgResult.Log, events))
 			err = msgResult.Error
+			events = nil
 			break
 		}
 
 		msgLogs = append(msgLogs,
 			fmt.Sprintf("msg:%d,success:%v,log:%s,events:%v",
 				i, true, msgResult.Log, events))
 	}
-	events = append(events, ctx.EventLogger().Events()...)
+
+	if err == nil {
+		events = append(events, ctx.EventLogger().Events()...)
```

Comment:
can't we have non-contract events here too?

Reply:
I don't think that is possible in this case; it initializes this event logger at the top of this method:
```go
ctx = ctx.WithEventLogger(NewEventLogger())
```